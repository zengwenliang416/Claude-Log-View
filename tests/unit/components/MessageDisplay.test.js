import { describe, it, expect, beforeEach, vi } from 'vitest'
import { mount } from '@vue/test-utils'
import { nextTick } from 'vue'
import MessageDisplay from '@/components/MainContent/MessageDisplay.vue'
import { createFilterTestData, createMessageSet } from '../../helpers/test-data.js'
import { mountComponent, waitForTicks } from '../../helpers/vue-test-utils.js'

describe('MessageDisplay.vue', () => {
  let wrapper
  const mockMessages = createFilterTestData()

  describe('Default Display Behavior', () => {
    beforeEach(() => {
      wrapper = mountComponent(MessageDisplay, {
        props: {
          messages: mockMessages,
          currentIndex: 0,
          loading: false,
          error: null
        }
      })
    })

    it('should display all messages by default without filters', () => {
      const messageItems = wrapper.findAll('.message-item')
      expect(messageItems).toHaveLength(mockMessages.length)
    })

    it('should render messages in scrollable container', () => {
      const container = wrapper.find('.messages-container')
      expect(container.exists()).toBe(true)
      expect(container.element.style.overflowY).toBe('auto')
    })

    it('should show all message types with proper indicators', () => {
      const typeIndicators = wrapper.findAll('.message-type-indicator')
      expect(typeIndicators).toHaveLength(mockMessages.length)
      
      const labels = typeIndicators.map(indicator => 
        indicator.find('.type-label').text()
      )
      
      expect(labels).toContain('USER')
      expect(labels).toContain('ASSISTANT')
      expect(labels).toContain('TOOL (Bash)')
      expect(labels).toContain('TOOL RESULT')
      expect(labels).toContain('SUMMARY')
    })

    it('should apply correct CSS classes for message types', () => {
      const messageItems = wrapper.findAll('.message-item')
      
      // Check for proper type classes
      expect(messageItems.some(item => item.classes().includes('message-user'))).toBe(true)
      expect(messageItems.some(item => item.classes().includes('message-assistant'))).toBe(true)
      expect(messageItems.some(item => item.classes().includes('message-tool'))).toBe(true)
      expect(messageItems.some(item => item.classes().includes('message-tool-result'))).toBe(true)
      expect(messageItems.some(item => item.classes().includes('message-summary'))).toBe(true)
    })
  })

  describe('Type Indicators', () => {
    beforeEach(() => {
      wrapper = mountComponent(MessageDisplay, {
        props: {
          messages: mockMessages,
          currentIndex: 0
        }
      })
    })

    it('should position type indicators in upper left corner', () => {
      const indicators = wrapper.findAll('.message-type-indicator')
      
      indicators.forEach(indicator => {
        const styles = getComputedStyle(indicator.element)
        expect(indicator.element.style.position).toBe('absolute')
        expect(indicator.element.style.top).toBeDefined()
        expect(indicator.element.style.left).toBeDefined()
        expect(indicator.element.style.zIndex).toBe('10')
      })
    })

    it('should display color-coded badges correctly', () => {
      const userMessage = wrapper.find('.message-user .type-label')
      const assistantMessage = wrapper.find('.message-assistant .type-label')
      const toolMessage = wrapper.find('.message-tool .type-label')
      const toolResultMessage = wrapper.find('.message-tool-result .type-label')
      const summaryMessage = wrapper.find('.message-summary .type-label')

      expect(userMessage.exists()).toBe(true)
      expect(assistantMessage.exists()).toBe(true)
      expect(toolMessage.exists()).toBe(true)
      expect(toolResultMessage.exists()).toBe(true)
      expect(summaryMessage.exists()).toBe(true)
    })

    it('should show tool names in tool type indicators', () => {
      const toolIndicators = wrapper.findAll('.message-tool .type-label')
      
      const toolTexts = toolIndicators.map(indicator => indicator.text())
      expect(toolTexts.some(text => text.includes('TOOL (Bash)'))).toBe(true)
      expect(toolTexts.some(text => text.includes('TOOL (Read)'))).toBe(true)
    })
  })

  describe('Scrollable Display', () => {
    beforeEach(() => {
      const largeMessageSet = createMessageSet(50)
      wrapper = mountComponent(MessageDisplay, {
        props: {
          messages: largeMessageSet,
          currentIndex: 25
        }
      })
    })

    it('should handle large message datasets efficiently', () => {
      const messageItems = wrapper.findAll('.message-item')
      expect(messageItems).toHaveLength(50)
      
      const container = wrapper.find('.messages-container')
      expect(container.exists()).toBe(true)
    })

    it('should implement smooth scrolling behavior', async () => {
      const container = wrapper.find('.messages-container')
      expect(container.classes()).toContain('messages-container')
      
      // Check for smooth scroll CSS property
      const styles = getComputedStyle(container.element)
      expect(container.element.style.scrollBehavior).toBe('smooth')
    })

    it('should auto-scroll to current message on index change', async () => {
      const scrollIntoViewSpy = vi.fn()
      
      // Mock scrollIntoView
      Element.prototype.scrollIntoView = scrollIntoViewSpy
      
      await wrapper.setProps({ currentIndex: 10 })
      await nextTick()
      
      expect(scrollIntoViewSpy).toHaveBeenCalledWith({
        behavior: 'smooth',
        block: 'center'
      })
    })

    it('should highlight active message correctly', async () => {
      await wrapper.setProps({ currentIndex: 5 })
      await waitForTicks(2)
      
      const activeMessage = wrapper.find('.message-active')
      expect(activeMessage.exists()).toBe(true)
      expect(activeMessage.classes()).toContain('message-active')
    })
  })

  describe('Navigation Integration', () => {
    beforeEach(() => {
      wrapper = mountComponent(MessageDisplay, {
        props: {
          messages: mockMessages,
          currentIndex: 2
        }
      })
    })

    it('should synchronize with sidebar navigation', async () => {
      const initialActiveMessage = wrapper.find('.message-active')
      expect(initialActiveMessage.exists()).toBe(true)
      
      // Change current index (simulating sidebar click)
      await wrapper.setProps({ currentIndex: 4 })
      await nextTick()
      
      const newActiveMessage = wrapper.find('.message-active')
      expect(newActiveMessage.exists()).toBe(true)
      
      // Verify the active message changed
      const messageItems = wrapper.findAll('.message-item')
      expect(messageItems[4].classes()).toContain('message-active')
    })

    it('should handle index changes gracefully', async () => {
      // Test boundary conditions
      await wrapper.setProps({ currentIndex: 0 })
      expect(wrapper.find('.message-active').exists()).toBe(true)
      
      await wrapper.setProps({ currentIndex: mockMessages.length - 1 })
      expect(wrapper.find('.message-active').exists()).toBe(true)
      
      // Test invalid index
      await wrapper.setProps({ currentIndex: mockMessages.length + 10 })
      // Should not crash and should handle gracefully
      expect(wrapper.find('.message-item')).toBeTruthy()
    })
  })

  describe('Performance Optimizations', () => {
    it('should handle 607+ messages efficiently', async () => {
      const largeDataset = createMessageSet(607)
      
      const startTime = performance.now()
      
      wrapper = mountComponent(MessageDisplay, {
        props: {
          messages: largeDataset,
          currentIndex: 0
        }
      })
      
      await nextTick()
      
      const endTime = performance.now()
      const renderTime = endTime - startTime
      
      // Should render within reasonable time (less than 100ms)
      expect(renderTime).toBeLessThan(100)
      expect(wrapper.findAll('.message-item')).toHaveLength(607)
    })

    it('should implement efficient re-rendering on prop changes', async () => {
      const renderSpy = vi.spyOn(wrapper.vm, '$forceUpdate')
      
      await wrapper.setProps({ currentIndex: 1 })
      await wrapper.setProps({ currentIndex: 2 })
      await wrapper.setProps({ currentIndex: 3 })
      
      // Should not force excessive re-renders
      expect(renderSpy).not.toHaveBeenCalled()
    })
  })

  describe('Loading and Error States', () => {
    it('should display loading state correctly', () => {
      wrapper = mountComponent(MessageDisplay, {
        props: {
          messages: [],
          loading: true,
          error: null
        }
      })

      const loadingState = wrapper.find('.loading-state')
      expect(loadingState.exists()).toBe(true)
      expect(loadingState.text()).toContain('Loading messages...')
      
      const spinner = wrapper.findComponent({ name: 'LoadingSpinner' })
      expect(spinner.exists()).toBe(true)
    })

    it('should display error state correctly', () => {
      const errorMessage = 'Failed to load messages'
      wrapper = mountComponent(MessageDisplay, {
        props: {
          messages: [],
          loading: false,
          error: errorMessage
        }
      })

      const errorState = wrapper.find('.error-state')
      expect(errorState.exists()).toBe(true)
      expect(errorState.text()).toContain('Error Loading Messages')
      expect(errorState.text()).toContain(errorMessage)
    })

    it('should display empty state correctly', () => {
      wrapper = mountComponent(MessageDisplay, {
        props: {
          messages: [],
          loading: false,
          error: null
        }
      })

      const emptyState = wrapper.find('.empty-state')
      expect(emptyState.exists()).toBe(true)
      expect(emptyState.text()).toContain('No Messages Available')
    })
  })

  describe('Accessibility', () => {
    beforeEach(() => {
      wrapper = mountComponent(MessageDisplay, {
        props: {
          messages: mockMessages,
          currentIndex: 0
        }
      })
    })

    it('should have proper ARIA attributes', () => {
      const main = wrapper.find('main')
      expect(main.attributes('class')).toBe('message-display')
      
      const activeMessage = wrapper.find('.message-active')
      expect(activeMessage.exists()).toBe(true)
    })

    it('should support keyboard navigation', async () => {
      const messagesContainer = wrapper.find('.messages-container')
      expect(messagesContainer.exists()).toBe(true)
      
      // Container should be focusable for keyboard navigation
      expect(messagesContainer.element.tabIndex).toBeGreaterThanOrEqual(-1)
    })

    it('should provide meaningful content for screen readers', () => {
      const typeLabels = wrapper.findAll('.type-label')
      
      typeLabels.forEach(label => {
        expect(label.text()).toMatch(/^(USER|ASSISTANT|TOOL|TOOL RESULT|SUMMARY)/)
      })
    })
  })

  describe('Responsive Design', () => {
    it('should adapt to mobile viewports', () => {
      // Test mobile-specific styles are applied
      expect(wrapper.html()).toMatchSnapshot()
    })

    it('should handle different screen sizes', () => {
      const container = wrapper.find('.messages-container')
      expect(container.exists()).toBe(true)
      
      // Check that responsive classes can be applied
      expect(wrapper.classes()).not.toContain('desktop-only')
    })
  })

  describe('Content Caching', () => {
    it('should handle message updates efficiently', async () => {
      const initialMessages = createMessageSet(10)
      wrapper = mountComponent(MessageDisplay, {
        props: {
          messages: initialMessages,
          currentIndex: 0
        }
      })

      const updatedMessages = [...initialMessages, ...createMessageSet(5)]
      
      await wrapper.setProps({ messages: updatedMessages })
      await nextTick()
      
      expect(wrapper.findAll('.message-item')).toHaveLength(15)
    })
  })

  afterEach(() => {
    if (wrapper) {
      wrapper.unmount()
    }
  })
})