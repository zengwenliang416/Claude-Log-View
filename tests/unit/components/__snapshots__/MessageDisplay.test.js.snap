// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MessageDisplay.vue > Responsive Design > should adapt to mobile viewports 1`] = `
"<main data-v-eba6b37b="" class="message-display">
  <div data-v-eba6b37b="" class="messages-container">
    <div data-v-eba6b37b="" class="message-item message-active message-user">
      <!-- Message Type Indicator -->
      <div data-v-eba6b37b="" class="message-type-indicator"><span data-v-eba6b37b="" class="type-label">USER</span></div><!-- Message Content -->
      <div data-v-eba6b37b="" class="message-wrapper">
        <header data-v-bae21dd0="" data-v-eba6b37b="" class="message-header">
          <div data-v-bae21dd0="" class="header-main">
            <div data-v-bae21dd0="" class="message-type-badge message-user">USER</div>
            <div data-v-bae21dd0="" class="message-meta"><span data-v-bae21dd0="" class="message-timestamp">2025/8/6 14:58:25</span><span data-v-bae21dd0="" class="message-uuid">1</span></div>
          </div>
          <!--v-if-->
          <div data-v-bae21dd0="" class="header-actions"><button data-v-bae21dd0="" class="toggle-extra"> Show Details </button></div>
        </header>
        <div data-v-51ef8b4e="" data-v-eba6b37b="" class="message-content">
          <!-- Summary Message -->
          <!-- User/Assistant Messages -->
          <div data-v-51ef8b4e="" class="content-section">
            <h3 data-v-51ef8b4e="" class="section-title">Assistant Response</h3>
            <div data-v-51ef8b4e="" class="message-parts">
              <div data-v-51ef8b4e="" class="message-part text">
                <!-- Text Content (Plain Text) -->
                <div data-v-51ef8b4e="" class="text-content"><pre data-v-51ef8b4e="" class="text-block">Hello world</pre>
                </div>
              </div>
            </div>
          </div><!-- Metadata Section -->
          <!--v-if-->
          <!-- Actions -->
          <div data-v-51ef8b4e="" class="content-actions"><button data-v-51ef8b4e="" class="action-button">Show Raw Data </button><button data-v-51ef8b4e="" class="action-button"> Copy Message </button></div>
        </div>
      </div>
    </div>
    <div data-v-eba6b37b="" class="message-item message-assistant">
      <!-- Message Type Indicator -->
      <div data-v-eba6b37b="" class="message-type-indicator"><span data-v-eba6b37b="" class="type-label">ASSISTANT</span></div><!-- Message Content -->
      <div data-v-eba6b37b="" class="message-wrapper">
        <header data-v-bae21dd0="" data-v-eba6b37b="" class="message-header">
          <div data-v-bae21dd0="" class="header-main">
            <div data-v-bae21dd0="" class="message-type-badge message-assistant">ASSISTANT</div>
            <div data-v-bae21dd0="" class="message-meta"><span data-v-bae21dd0="" class="message-timestamp">2025/8/6 14:58:25</span><span data-v-bae21dd0="" class="message-uuid">2</span></div>
          </div>
          <!--v-if-->
          <div data-v-bae21dd0="" class="header-actions"><button data-v-bae21dd0="" class="toggle-extra"> Show Details </button></div>
        </header>
        <div data-v-51ef8b4e="" data-v-eba6b37b="" class="message-content">
          <!-- Summary Message -->
          <!-- User/Assistant Messages -->
          <div data-v-51ef8b4e="" class="content-section">
            <h3 data-v-51ef8b4e="" class="section-title">Assistant Response</h3>
            <div data-v-51ef8b4e="" class="message-parts">
              <div data-v-51ef8b4e="" class="message-part text">
                <!-- Text Content (Plain Text) -->
                <div data-v-51ef8b4e="" class="text-content"><pre data-v-51ef8b4e="" class="text-block">Hi there, how can I help?</pre>
                </div>
              </div>
            </div>
          </div><!-- Metadata Section -->
          <!--v-if-->
          <!-- Actions -->
          <div data-v-51ef8b4e="" class="content-actions"><button data-v-51ef8b4e="" class="action-button">Show Raw Data </button><button data-v-51ef8b4e="" class="action-button"> Copy Message </button></div>
        </div>
      </div>
    </div>
    <div data-v-eba6b37b="" class="message-item message-tool">
      <!-- Message Type Indicator -->
      <div data-v-eba6b37b="" class="message-type-indicator"><span data-v-eba6b37b="" class="type-label">TOOL (Bash)</span></div><!-- Message Content -->
      <div data-v-eba6b37b="" class="message-wrapper">
        <header data-v-bae21dd0="" data-v-eba6b37b="" class="message-header">
          <div data-v-bae21dd0="" class="header-main">
            <div data-v-bae21dd0="" class="message-type-badge message-tool">TOOL (Bash)</div>
            <div data-v-bae21dd0="" class="message-meta"><span data-v-bae21dd0="" class="message-timestamp">2025/8/6 14:58:25</span><span data-v-bae21dd0="" class="message-uuid">3</span></div>
          </div>
          <div data-v-bae21dd0="" class="header-tools"><span data-v-bae21dd0="" class="tools-label">Tools:</span>
            <div data-v-bae21dd0="" class="tool-badges"><span data-v-bae21dd0="" class="tool-badge">Bash</span></div>
          </div>
          <div data-v-bae21dd0="" class="header-actions"><button data-v-bae21dd0="" class="toggle-extra"> Show Details </button></div>
        </header>
        <div data-v-51ef8b4e="" data-v-eba6b37b="" class="message-content">
          <!-- Summary Message -->
          <!-- User/Assistant Messages -->
          <div data-v-51ef8b4e="" class="content-section">
            <h3 data-v-51ef8b4e="" class="section-title">Assistant Response</h3>
            <div data-v-51ef8b4e="" class="message-parts">
              <div data-v-51ef8b4e="" class="message-part text">
                <!-- Text Content (Plain Text) -->
                <div data-v-51ef8b4e="" class="text-content"><pre data-v-51ef8b4e="" class="text-block">Using Bash tool</pre>
                </div>
              </div>
              <div data-v-51ef8b4e="" class="message-part tool_use">
                <!-- Text Content (Plain Text) -->
                <!-- Tool Use -->
                <div data-v-51ef8b4e="" class="tool-use-content">
                  <div data-v-51ef8b4e="" class="tool-header"><span data-v-51ef8b4e="" class="tool-name">Bash</span><span data-v-51ef8b4e="" class="tool-id"></span></div>
                  <div data-v-51ef8b4e="" class="tool-parameters">
                    <h4 data-v-51ef8b4e="" class="parameters-title">Parameters:</h4><pre data-v-51ef8b4e="" class="code-block"><span class="hljs">[object Object]</span></pre>
                  </div>
                </div>
              </div>
            </div>
          </div><!-- Metadata Section -->
          <!--v-if-->
          <!-- Actions -->
          <div data-v-51ef8b4e="" class="content-actions"><button data-v-51ef8b4e="" class="action-button">Show Raw Data </button><button data-v-51ef8b4e="" class="action-button"> Copy Message </button></div>
        </div>
      </div>
    </div>
    <div data-v-eba6b37b="" class="message-item message-tool-result">
      <!-- Message Type Indicator -->
      <div data-v-eba6b37b="" class="message-type-indicator"><span data-v-eba6b37b="" class="type-label">TOOL RESULT</span></div><!-- Message Content -->
      <div data-v-eba6b37b="" class="message-wrapper">
        <header data-v-bae21dd0="" data-v-eba6b37b="" class="message-header">
          <div data-v-bae21dd0="" class="header-main">
            <div data-v-bae21dd0="" class="message-type-badge message-tool-result">TOOL RESULT</div>
            <div data-v-bae21dd0="" class="message-meta"><span data-v-bae21dd0="" class="message-timestamp">2025/8/6 14:58:25</span><span data-v-bae21dd0="" class="message-uuid">4</span></div>
          </div>
          <div data-v-bae21dd0="" class="header-tools"><span data-v-bae21dd0="" class="tools-label">Tools:</span>
            <div data-v-bae21dd0="" class="tool-badges"><span data-v-bae21dd0="" class="tool-badge">Bash</span></div>
          </div>
          <div data-v-bae21dd0="" class="header-actions"><button data-v-bae21dd0="" class="toggle-extra"> Show Details </button></div>
        </header>
        <div data-v-51ef8b4e="" data-v-eba6b37b="" class="message-content">
          <!-- Summary Message -->
          <!-- Tool Result Messages -->
          <div data-v-51ef8b4e="" class="content-section">
            <h3 data-v-51ef8b4e="" class="section-title">Tool Result</h3>
            <div data-v-51ef8b4e="" class="tool-result-content">
              <div data-v-51ef8b4e="" class="result-header"><span data-v-51ef8b4e="" class="result-type"> Type: Bash</span></div>
              <div data-v-51ef8b4e="" class="result-content">
                <h4 data-v-51ef8b4e="" class="result-title">Output:</h4>
                <div data-v-51ef8b4e="" class="result-data"><pre data-v-51ef8b4e="" class="text-block">file1.txt
file2.txt</pre>
                </div>
              </div>
            </div>
          </div><!-- Metadata Section -->
          <!--v-if-->
          <!-- Actions -->
          <div data-v-51ef8b4e="" class="content-actions"><button data-v-51ef8b4e="" class="action-button">Show Raw Data </button><button data-v-51ef8b4e="" class="action-button"> Copy Message </button></div>
        </div>
      </div>
    </div>
    <div data-v-eba6b37b="" class="message-item message-tool">
      <!-- Message Type Indicator -->
      <div data-v-eba6b37b="" class="message-type-indicator"><span data-v-eba6b37b="" class="type-label">TOOL (Read)</span></div><!-- Message Content -->
      <div data-v-eba6b37b="" class="message-wrapper">
        <header data-v-bae21dd0="" data-v-eba6b37b="" class="message-header">
          <div data-v-bae21dd0="" class="header-main">
            <div data-v-bae21dd0="" class="message-type-badge message-tool">TOOL (Read)</div>
            <div data-v-bae21dd0="" class="message-meta"><span data-v-bae21dd0="" class="message-timestamp">2025/8/6 14:58:25</span><span data-v-bae21dd0="" class="message-uuid">5</span></div>
          </div>
          <div data-v-bae21dd0="" class="header-tools"><span data-v-bae21dd0="" class="tools-label">Tools:</span>
            <div data-v-bae21dd0="" class="tool-badges"><span data-v-bae21dd0="" class="tool-badge">Read</span></div>
          </div>
          <div data-v-bae21dd0="" class="header-actions"><button data-v-bae21dd0="" class="toggle-extra"> Show Details </button></div>
        </header>
        <div data-v-51ef8b4e="" data-v-eba6b37b="" class="message-content">
          <!-- Summary Message -->
          <!-- User/Assistant Messages -->
          <div data-v-51ef8b4e="" class="content-section">
            <h3 data-v-51ef8b4e="" class="section-title">Assistant Response</h3>
            <div data-v-51ef8b4e="" class="message-parts">
              <div data-v-51ef8b4e="" class="message-part text">
                <!-- Text Content (Plain Text) -->
                <div data-v-51ef8b4e="" class="text-content"><pre data-v-51ef8b4e="" class="text-block">Using Read tool</pre>
                </div>
              </div>
              <div data-v-51ef8b4e="" class="message-part tool_use">
                <!-- Text Content (Plain Text) -->
                <!-- Tool Use -->
                <div data-v-51ef8b4e="" class="tool-use-content">
                  <div data-v-51ef8b4e="" class="tool-header"><span data-v-51ef8b4e="" class="tool-name">Read</span><span data-v-51ef8b4e="" class="tool-id"></span></div>
                  <div data-v-51ef8b4e="" class="tool-parameters">
                    <h4 data-v-51ef8b4e="" class="parameters-title">Parameters:</h4><pre data-v-51ef8b4e="" class="code-block"><span class="hljs">[object Object]</span></pre>
                  </div>
                </div>
              </div>
            </div>
          </div><!-- Metadata Section -->
          <!--v-if-->
          <!-- Actions -->
          <div data-v-51ef8b4e="" class="content-actions"><button data-v-51ef8b4e="" class="action-button">Show Raw Data </button><button data-v-51ef8b4e="" class="action-button"> Copy Message </button></div>
        </div>
      </div>
    </div>
    <div data-v-eba6b37b="" class="message-item message-tool-result">
      <!-- Message Type Indicator -->
      <div data-v-eba6b37b="" class="message-type-indicator"><span data-v-eba6b37b="" class="type-label">TOOL RESULT</span></div><!-- Message Content -->
      <div data-v-eba6b37b="" class="message-wrapper">
        <header data-v-bae21dd0="" data-v-eba6b37b="" class="message-header">
          <div data-v-bae21dd0="" class="header-main">
            <div data-v-bae21dd0="" class="message-type-badge message-tool-result">TOOL RESULT</div>
            <div data-v-bae21dd0="" class="message-meta"><span data-v-bae21dd0="" class="message-timestamp">2025/8/6 14:58:25</span><span data-v-bae21dd0="" class="message-uuid">6</span></div>
          </div>
          <div data-v-bae21dd0="" class="header-tools"><span data-v-bae21dd0="" class="tools-label">Tools:</span>
            <div data-v-bae21dd0="" class="tool-badges"><span data-v-bae21dd0="" class="tool-badge">Read</span></div>
          </div>
          <div data-v-bae21dd0="" class="header-actions"><button data-v-bae21dd0="" class="toggle-extra"> Show Details </button></div>
        </header>
        <div data-v-51ef8b4e="" data-v-eba6b37b="" class="message-content">
          <!-- Summary Message -->
          <!-- Tool Result Messages -->
          <div data-v-51ef8b4e="" class="content-section">
            <h3 data-v-51ef8b4e="" class="section-title">Tool Result</h3>
            <div data-v-51ef8b4e="" class="tool-result-content">
              <div data-v-51ef8b4e="" class="result-header"><span data-v-51ef8b4e="" class="result-type"> Type: Read</span></div>
              <div data-v-51ef8b4e="" class="result-content">
                <h4 data-v-51ef8b4e="" class="result-title">Output:</h4>
                <div data-v-51ef8b4e="" class="result-data"><pre data-v-51ef8b4e="" class="text-block">File content here</pre>
                </div>
              </div>
            </div>
          </div><!-- Metadata Section -->
          <!--v-if-->
          <!-- Actions -->
          <div data-v-51ef8b4e="" class="content-actions"><button data-v-51ef8b4e="" class="action-button">Show Raw Data </button><button data-v-51ef8b4e="" class="action-button"> Copy Message </button></div>
        </div>
      </div>
    </div>
    <div data-v-eba6b37b="" class="message-item message-user">
      <!-- Message Type Indicator -->
      <div data-v-eba6b37b="" class="message-type-indicator"><span data-v-eba6b37b="" class="type-label">USER</span></div><!-- Message Content -->
      <div data-v-eba6b37b="" class="message-wrapper">
        <header data-v-bae21dd0="" data-v-eba6b37b="" class="message-header">
          <div data-v-bae21dd0="" class="header-main">
            <div data-v-bae21dd0="" class="message-type-badge message-user">USER</div>
            <div data-v-bae21dd0="" class="message-meta"><span data-v-bae21dd0="" class="message-timestamp">2025/8/6 14:58:25</span><span data-v-bae21dd0="" class="message-uuid">7</span></div>
          </div>
          <!--v-if-->
          <div data-v-bae21dd0="" class="header-actions"><button data-v-bae21dd0="" class="toggle-extra"> Show Details </button></div>
        </header>
        <div data-v-51ef8b4e="" data-v-eba6b37b="" class="message-content">
          <!-- Summary Message -->
          <!-- User/Assistant Messages -->
          <div data-v-51ef8b4e="" class="content-section">
            <h3 data-v-51ef8b4e="" class="section-title">Assistant Response</h3>
            <div data-v-51ef8b4e="" class="message-parts">
              <div data-v-51ef8b4e="" class="message-part text">
                <!-- Text Content (Plain Text) -->
                <div data-v-51ef8b4e="" class="text-content"><pre data-v-51ef8b4e="" class="text-block">Thank you for the help</pre>
                </div>
              </div>
            </div>
          </div><!-- Metadata Section -->
          <!--v-if-->
          <!-- Actions -->
          <div data-v-51ef8b4e="" class="content-actions"><button data-v-51ef8b4e="" class="action-button">Show Raw Data </button><button data-v-51ef8b4e="" class="action-button"> Copy Message </button></div>
        </div>
      </div>
    </div>
    <div data-v-eba6b37b="" class="message-item message-summary">
      <!-- Message Type Indicator -->
      <div data-v-eba6b37b="" class="message-type-indicator"><span data-v-eba6b37b="" class="type-label">SUMMARY</span></div><!-- Message Content -->
      <div data-v-eba6b37b="" class="message-wrapper">
        <header data-v-bae21dd0="" data-v-eba6b37b="" class="message-header">
          <div data-v-bae21dd0="" class="header-main">
            <div data-v-bae21dd0="" class="message-type-badge message-summary">SUMMARY</div>
            <div data-v-bae21dd0="" class="message-meta"><span data-v-bae21dd0="" class="message-timestamp">2025/8/6 14:58:25</span><span data-v-bae21dd0="" class="message-uuid">8</span></div>
          </div>
          <!--v-if-->
          <div data-v-bae21dd0="" class="header-actions"><button data-v-bae21dd0="" class="toggle-extra"> Show Details </button></div>
        </header>
        <div data-v-51ef8b4e="" data-v-eba6b37b="" class="message-content">
          <!-- Summary Message -->
          <div data-v-51ef8b4e="" class="content-section">
            <h3 data-v-51ef8b4e="" class="section-title">Summary</h3>
            <div data-v-51ef8b4e="" class="summary-content"><pre data-v-51ef8b4e="" class="text-block">Task completed successfully</pre>
            </div>
          </div><!-- Metadata Section -->
          <!--v-if-->
          <!-- Actions -->
          <div data-v-51ef8b4e="" class="content-actions"><button data-v-51ef8b4e="" class="action-button">Show Raw Data </button><button data-v-51ef8b4e="" class="action-button"> Copy Message </button></div>
        </div>
      </div>
    </div>
  </div>
</main>"
`;
